compositor_render/src/transformations/text_renderer.rs

```diff
@@ -0,0 +1,197 @@
+use compositor_common::scene::TextParams;
+use glyphon::{
+    AttrsOwned, Buffer, Color, FontSystem, Metrics, Shaping, SwashCache, TextArea, TextAtlas,
+    TextBounds,
+};
+use log::info;
+use wgpu::{
+    CommandEncoderDescriptor, LoadOp, MultisampleState, Operations, RenderPassColorAttachment,
+    RenderPassDescriptor, TextureFormat,
+};
+
+use crate::renderer::{texture::NodeTexture, RenderCtx};
+
+#[allow(dead_code)]
+pub struct TextSpec {
+    content: String,
+    attributes: AttrsOwned,
+    font_size: f32,
+    line_height: f32,
+}
+
+impl From<TextParams> for TextSpec {
+    fn from(text_params: TextParams) -> Self {
+        let attributes = Self::get_attrs_owned(&text_params);
+        let font_size = text_params.font_size;
+        let line_height = text_params.line_height.unwrap_or(font_size);
+
+        Self {
+            content: text_params.content,
+            attributes,
+            font_size,
+            line_height,
+        }
+    }
+}
+
+impl TextSpec {
+    fn get_attrs_owned(text_params: &TextParams) -> AttrsOwned {
+        let color = match text_params.color_rgba {
+            Some((r, g, b, a)) => glyphon::Color::rgba(r, g, b, a),
+            None => glyphon::Color::rgb(255, 255, 255),
+        };
+
+        let family = match &text_params.font_family {
+            Some(font_family_name) => glyphon::FamilyOwned::Name(font_family_name.clone()),
+            None => glyphon::FamilyOwned::SansSerif,
+        };
+
+        let style = match text_params.style {
+            Some(compositor_common::scene::Style::Normal) | None => glyphon::Style::Normal,
+            Some(compositor_common::scene::Style::Italic) => glyphon::Style::Italic,
+            Some(compositor_common::scene::Style::Oblique) => glyphon::Style::Oblique,
+        };
+
+        AttrsOwned {
+            color_opt: Some(color),
+            family_owned: family,
+            stretch: Default::default(),
+            style,
+            weight: Default::default(),
+            metadata: Default::default(),
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct TextRendererCtx {
+    font_system: FontSystem,
+}
+
+impl TextRendererCtx {
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        Self {
+            font_system: FontSystem::new(),
+        }
+    }
+}
+
+impl Default for TextRendererCtx {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[allow(dead_code)]
+pub struct TextRenderer {
+    was_rendered: bool,
```

Comment:
```suggestion
    was_rendered: Mutex<bool>,
```
maybe? It would make for a cleaner api